core/Dockerfile

```
@@ -1,3 +1,27 @@
+### Build train-sim dynamic lib (not used in core for now, but eventually will)
+# Will be used for testing
+FROM rust:latest AS train_sim_rust_env
+WORKDIR /app
+
+RUN rustup component add llvm-tools && \
+    rustup component add rustfmt && \
+    rustup component add clippy && \
+    cargo install --locked grcov
+ENV RUSTFLAGS="-Cinstrument-coverage"
+ENV LLVM_PROFILE_FILE="train-sim-%p-%m.profraw"
+
+COPY --from=train_sim_dir . .
+
+RUN cargo build -p uniffi-bindgen --release
```

minor: I'm all for long option instead of short one, we usually read them much more than we write them, so I'd say readibility should be preferred to conciseness.
```suggestion
RUN cargo build --package uniffi-bindgen --release
```